packages/angular-output-target/src/generate-angular-component.ts

```diff
@@ -1,132 +1,141 @@
-        }
-      });
-    });
+  const hasInputs = inputs.length > 0;
+  const hasOutputs = outputs.length > 0;
+  const hasMethods = methods.length > 0;
 
-    const componentEvents: string[] = [
-      '', // Empty first line
-    ];
+  // Formats the input strings into comma separated, single quoted values.
+  const formattedInputs = inputs.map((input) => `'${input}'`).join(', ');
+  // Formats the output strings into comma separated, single quoted values.
+  const formattedOutputs = outputs.map((output) => `'${output}'`).join(', ');
+  // Formats the method strings into comma separated, single quoted values.
+  const formattedMethods = methods.map((method) => `'${method}'`).join(', ');
 
-    // Generate outputs
-    outputs.forEach((output, index) => {
-      componentEvents.push(
-        `  /**
-   * ${output.docs.text} ${output.docs.tags.map((tag) => `@${tag.name} ${tag.text}`)}
-   */`
-      );
-      /**
-       * The original attribute contains the original type defined by the devs.
-       * This regexp normalizes the reference, by removing linebreaks,
-       * replacing consecutive spaces with a single space, and adding a single space after commas.
-       **/
+  const proxyCmpOptions = [];
 
-      const outputTypeRemapped = Object.entries(outputReferenceRemap).reduce(
-        (type, [src, dst]) => {
-          return type
-            .replace(new RegExp(`^${src}$`, 'g'), `${dst}`)
-            .replace(new RegExp(`([^\\w])${src}([^\\w])`, 'g'), (v, p1, p2) => [p1, dst, p2].join(''));
-        },
-        output.complexType.original
-          .replace(/\n/g, ' ')
-          .replace(/\s{2,}/g, ' ')
-          .replace(/,\s*/g, ', ')
-      );
-      componentEvents.push(`  ${output.name}: EventEmitter<CustomEvent<${outputTypeRemapped.trim()}>>;`);
+  if (includeImportCustomElements && defineCustomElementFn) {
+    proxyCmpOptions.push(`\n  defineCustomElementFn: ${defineCustomElementFn}`);
+  }
 
-      if (index === outputs.length - 1) {
-        // Empty line to push end `}` to new line
-        componentEvents.push('\n');
-      }
-    });
+  if (hasInputs) {
+    proxyCmpOptions.push(`\n  inputs: [${formattedInputs}]`);
+  }
 
-    const lines = [
-      '', // Empty first line
-      `${[...outputsInterface].join('\n')}
-export declare interface ${tagNameAsPascal} extends Components.${tagNameAsPascal} {${
-        componentEvents.length > 1 ? componentEvents.join('\n') : ''
-      }}
+  if (hasMethods) {
+    proxyCmpOptions.push(`\n  methods: [${formattedMethods}]`);
+  }
 
-${getProxyCmp(cmpMeta.tagName, includeImportCustomElements, inputs, methods)}
+  const output = `@ProxyCmp({${proxyCmpOptions.join(',')}\n})
 @Component({
-  ${directiveOpts.join(',\n  ')}
+  selector: '${tagName}',
+  template: '<ng-content></ng-content>',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  // eslint-disable-next-line @angular-eslint/no-inputs-metadata-property
```

Comment:
To be honest I don't think it makes sense to document why to the consumer. Angular warns you shouldn't use this pattern because you should be using the `@Input()` decorated approach. We don't do that because we would be re-assigning class members and would be overwriting the definition on the web component. This means to do it "semantically" correct, we would need to create inputs with default values/types that match the WC. 

The generated proxy output is a nightmare of lint errors/warnings. We disable all of them at the top of the file, but moving to inline disabling is valuable over time.